compact.go

```diff
@@ -593,9 +616,20 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, meta *BlockMeta,
 		allSymbols = make(map[string]struct{}, 1<<16)
 		closers    = []io.Closer{}
 	)
-	defer func() { closeAll(closers...) }()
+	defer func() {
+		closeAll(closers...)
+		c.metrics.populatingBlocks.Add(-1)
+	}()
+
+	c.metrics.populatingBlocks.Inc()
```

Comment:
As the description says set to 1, I would prefer explicitly using the `Set` function.